test/test_nested_tensor_masking.py

```diff
@@ -1,350 +1,976 @@
+        ])
+
+        expected_nt3 = nt.nested_tensor([
+                nt.nested_tensor([
+                    nt.nested_tensor([
+                        torch.tensor([[1, 2, 3, 4],
+                                      [5, 6, 7, 8]])
+                    ]),
+                    nt.nested_tensor([
+                        torch.tensor([[0, 0], 
+                                      [3, 4]])
+                    ]),
+                    nt.nested_tensor([
+                        torch.tensor([[1]])
+                    ]),
+                ])
+            ])
+
+        expected_nt4 = nt.nested_tensor([
             nt.nested_tensor([
-                torch.tensor([[[5, 6],
-                               [7, 8]]]),
-            ]),
+                nt.nested_tensor([
+                    nt.nested_tensor([
+                        torch.tensor([1, 2, 3, 4]),
+                        torch.tensor([5, 6, 7, 8])
+                    ])
+                ]),
+                nt.nested_tensor([
+                        nt.nested_tensor([
+                            torch.tensor([0, 0]),
+                            torch.tensor([3, 4])
+                        ])
+                ]),
+                nt.nested_tensor([
+                        nt.nested_tensor([
+                            torch.tensor([1]),
+                            torch.tensor([], dtype=torch.long)
+                        ])
+                ])
+            ])
         ])
 
-        tensor, mask = original_nt.to_tensor_mask(mask_dim=3)
+        expected_nt5 = nt.nested_tensor([
+            nt.nested_tensor([
+                nt.nested_tensor([
+                    nt.nested_tensor([
+                        nt.nested_tensor([
+                            torch.tensor(1),
+                            torch.tensor(2),
+                            torch.tensor(3),
+                            torch.tensor(4)
+                        ]),
+                        nt.nested_tensor([
+                            torch.tensor(5),
+                            torch.tensor(6),
+                            torch.tensor(7),
+                            torch.tensor(8)
+                        ]),
+                    ])
+                ]),
+                nt.nested_tensor([
+                        nt.nested_tensor([
+                            nt.nested_tensor([
+                                torch.tensor(0),
+                                torch.tensor(0)
+                            ]),
+                            nt.nested_tensor([
+                                torch.tensor(3),
+                                torch.tensor(4)
+                            ])
+                        ])
+                ]),
+                nt.nested_tensor([
+                        nt.nested_tensor([
+                            nt.nested_tensor([
+                                torch.tensor(1)
+                            ]),
+                            nt.nested_tensor([
```

Comment:
Let's talk about why you'd want this

Reply:
will follow up once to_nt is moved to C++